app/src/test/java/com/distillery/interview/ui/CurrentWeatherViewModelTests.kt

```diff
@@ -0,0 +1,75 @@
+package com.distillery.interview.ui
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import androidx.lifecycle.asFlow
+import com.distillery.interview.util.MainCoroutineRule
+import com.distillery.interview.data.models.Result
+import com.distillery.interview.data.models.WeatherResponse
+import com.distillery.interview.data.source.WeatherRepository
+import com.distillery.interview.ui.current_weather.CurrentWeatherViewModel
+import com.distillery.interview.util.getOrAwaitValue
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.whenever
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.test.TestCoroutineDispatcher
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+@ExperimentalCoroutinesApi
+class CurrentWeatherViewModelTests {
+
+    private val testCoroutineDispatcher = TestCoroutineDispatcher()
+
+    @get:Rule
+    val instantTaskExecutorRule = InstantTaskExecutorRule()
+
+    @get:Rule
+    val coroutineRule = MainCoroutineRule()
```

Comment:
We agreed on removing MainCoroutineRule component.

Reply:
We needed because it sets the main coroutines dispatcher to our test coroutine scope.